File: TeamCode/src/org/murraybridgebunyips/bunyipslib/tasks/bases/BunyipsTask.java
package org.murraybridgebunyips.bunyipslib.tasks.bases;

import androidx.annotation.NonNull;

import org.murraybridgebunyips.bunyipslib.BunyipsOpMode;
import org.murraybridgebunyips.bunyipslib.BunyipsSubsystem;

/**
 * Task interface with BunyipsOpMode dependency injection.
 */
public abstract class BunyipsTask extends Task {
    protected final BunyipsOpMode opMode;

    protected BunyipsTask(BunyipsOpMode opMode, double timeoutSeconds) {
        super(timeoutSeconds);
        this.opMode = opMode;
    }

    protected BunyipsTask(BunyipsOpMode opMode, double timeoutSeconds, @NonNull BunyipsSubsystem dependencySubsystem, boolean shouldOverrideConflictingTasks) {
        super(timeoutSeconds, dependencySubsystem, shouldOverrideConflictingTasks);
        this.opMode = opMode;
    }
}

File: TeamCode/src/org/murraybridgebunyips/bunyipslib/tasks/bases/BunyipsTask.kt
package org.murraybridgebunyips.bunyipslib.tasks.bases

import org.murraybridgebunyips.bunyipslib.BunyipsOpMode
import org.murraybridgebunyips.bunyipslib.BunyipsSubsystem

/**
 * Task interface with BunyipsOpMode dependency injection.
 */
abstract class BunyipsTask(open val opMode: BunyipsOpMode, time: Double) : Task(time) {
    constructor(opMode: BunyipsOpMode) : this(opMode, 0.0)
    constructor(
        opMode: BunyipsOpMode,
        time: Double,
        dependencySubsystem: BunyipsSubsystem,
        shouldOverrideConflictingTasks: Boolean
    ) : this(opMode, time) {
        addDependency(dependencySubsystem)
        overrideOnConflict = shouldOverrideConflictingTasks
    }
}